crates/witness-generator/src/rpc_generator.rs

```diff
@@ -32,55 +32,60 @@ impl RPCBlocksAndWitnessesBuilder {
     }
 
     /// Adds the provided HTTP headers to the RPC client.
-    pub fn with_headers<S: AsRef<str>>(mut self, headers: Vec<(S, S)>) -> Result<Self> {
-        self.header_map = headers
-            .into_iter()
-            .map(|(name, value)| {
-                Ok((
-                    HeaderName::from_str(name.as_ref())?,
-                    HeaderValue::from_str(value.as_ref())?,
-                ))
-            })
-            .collect::<Result<HeaderMap>>()?;
-        Ok(self)
+    ///
+    /// # Arguments
+    /// * `headers` - HTTP headers to include in RPC requests
+    pub fn with_headers(mut self, headers: HeaderMap) -> Self {
```

Comment:
Before the HTTP-headers configuration was quite messy; now it is simpler. More about it later.